va/caa_test.go

```diff
@@ -578,13 +575,400 @@ func TestIsCAAValidParams(t *testing.T) {
 	test.AssertError(t, err, "calling IsCAAValid without an AccountURIID")
 }
 
+var errCAABrokenDNSClient = errors.New("dnsClient is broken")
+
+// caaBrokenDNS implements the `dns.DNSClient` interface, but always returns
+// errors.
+type caaBrokenDNS struct{}
+
+func (b caaBrokenDNS) LookupTXT(_ context.Context, hostname string) ([]string, error) {
+	return nil, errCAABrokenDNSClient
+}
+
+func (b caaBrokenDNS) LookupHost(_ context.Context, hostname string) ([]net.IP, error) {
+	return nil, errCAABrokenDNSClient
+}
+
+func (b caaBrokenDNS) LookupCAA(_ context.Context, domain string) ([]*dns.CAA, string, error) {
+	return nil, "", errCAABrokenDNSClient
+}
+
+func TestDisabledMultiCAARechecking(t *testing.T) {
+	brokenRVA := setupRemote(nil, "broken", caaBrokenDNS{})
+	remoteVAs := []RemoteVA{{brokenRVA, "broken"}}
+	va, _ := setup(nil, 0, "local", remoteVAs, nil)
+
+	features.Set(features.Config{
+		EnforceMultiCAA:     false,
+		MultiCAAFullResults: false,
+	})
+	defer features.Reset()
+
+	isValidRes, err := va.IsCAAValid(context.TODO(), &vapb.IsCAAValidRequest{
+		Domain:           "present.com",
+		ValidationMethod: string(core.ChallengeTypeDNS01),
+		AccountURIID:     1,
+	})
+	test.AssertNotError(t, err, "Error during IsCAAValid")
+	// The primary VA can successfully recheck the CAA record and is allowed to
+	// issue for this domain. If `EnforceMultiCAA`` was enabled, the configured
+	// remote VA with broken dns.Client would fail the check and return a
+	// Problem, but that code path could never trigger.
+	test.AssertBoxedNil(t, isValidRes.Problem, "IsCAAValid returned a problem, but should not have")
+}
+
+// caaHijackedDNS implements the `dns.DNSClient` interface with a set of useful
+// test answers for CAA queries. It returns alternate CAA records than what
+// caaMockDNS returns simulating either a BGP hijack or DNS records that have
+// changed while queries were inflight.
+type caaHijackedDNS struct{}
+
+func (h caaHijackedDNS) LookupTXT(_ context.Context, hostname string) ([]string, error) {
+	return nil, nil
+}
+
+func (h caaHijackedDNS) LookupHost(_ context.Context, hostname string) ([]net.IP, error) {
+	ip := net.ParseIP("127.0.0.1")
+	return []net.IP{ip}, nil
+}
+func (h caaHijackedDNS) LookupCAA(_ context.Context, domain string) ([]*dns.CAA, string, error) {
+	// These records are altered from their caaMockDNS counterparts. Use this to
+	// tickle remoteValidationFailures.
+	var results []*dns.CAA
+	var record dns.CAA
+	switch strings.TrimRight(domain, ".") {
+	case "present.com", "present.servfail.com":
+		record.Tag = "issue"
+		record.Value = "www.honestachmed.dyndns.org"
```

Comment:
We don't reference https://bugzilla.mozilla.org/show_bug.cgi?id=647959 anywhere else in boulder yet, no need to start now. Any other value like "other-ca.com" would work just as well.

Reply:
Sure, I just thought it was silly.